browser/brave_profile_prefs.cc

```diff
@@ -102,21 +102,22 @@ void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(kHTTPSEVerywhereControlType, true);
   registry->RegisterBooleanPref(kNoScriptControlType, false);
   registry->RegisterBooleanPref(kAdControlType, true);
-  // > advanced view is defaulted to true for EXISTING users; false for new
-  bool is_new_user = false;
 
+  // advanced view is defaulted to true for EXISTING users; false for new.
+  // also see brave_profile_impl.cc for where default is locked in.
+  bool is_new_user = false;
 #if !defined(OS_ANDROID)
   is_new_user = first_run::IsChromeFirstRun();
 #endif
+  registry->RegisterBooleanPref(kShieldsAdvancedViewEnabled,
+                                is_new_user == false);
```

Comment:
Before, default was being set properly. but this doesn't actually set the profile value 🤦‍♂ My bad here when doing https://github.com/brave/brave-core/pull/3154

On next launch, this evaluates again and sets default to opposite. Since no user-recorded value exists, it was changing their default too 😢

Reply:
How about having advanced view related logic in `brave_profile_impl.cc` only?
In here, just calling `registry->RegisterBooleanPref(kShieldsAdvancedViewEnabled, false);`
and overwriting default could be done in `brave_profile_impl.cc`.
IMO, unifying related logic in one place seems better. WDYT?

Reply:
Great idea! Better to not have the default change too, I think